languages/purescript/exercises/concept/pattern-matching/.docs/introduction.md

```diff
@@ -0,0 +1,44 @@
+An `if/else` expression can be used to conditionally execute logic. PureScript also has another, more powerful way to conditionally execute logic: pattern matching. With pattern matching, a value can be tested against one or more _patterns_. An example of such a pattern is matching a value against a constant (e.g. `1` or `"hello"`).
```

Comment:
> An example of such a pattern is matching a value against a constant (e.g. `1` or `"hello"`).

Does PureScript have this? Your example does pattern matching through a series of conditions. I would have expected something like this if constant pattern matching was supported:

```purescript
describe number
  | 0 = "Zero"
  | 1 = "One"
```

If this is not supported, I think you can safely remove the bit about pattern matching about a constant.

Reply:
You can match constants like this:
```purs
describe 0 = "Zero"
describe 1 = "One"
describe n = "Something else"
-- the live above could omit the `n` as it's unused:
-- describe _ = "Something else"
```
or like this:
```purs
describe n = case n of
  0 -> "Zero"
  1 -> "One"
  _ -> "Something else"
  -- (or) x -> "Something else"
```
(which is the same as this):
```purs
describe = case _ of
  0 -> "Zero"
  1 -> "One"
  _ -> "Something else"
  -- (or) x -> "Something else"
```

Wildcards (the `_`) are a little weird in that while pattern matching they're catch-alls, but in a few special cases they create functions:
- `case _ of ...` becomes `\a -> case a of ...`
- `if _ then ...` becomes `\a -> if a then ...`
- `(_ + 2)` becomes `\a -> a + 2`
- `_ { foo = "bar" }` becomes `\a -> a { foo = "bar }`

Reply:
Also the catch-all I added to all these is necessary to avoid a compile error. You could define a partial pattern match which fails at runtime using `Partial`, but that's probably outside the scope of a beginner guide:
```purs
describe :: Partial => Int -> String
describe 0 = "Zero"
describe 1 = "One"
```

Using the function would then require `unsafePartial`:
```purs
unsafePartial describe 2
```

Reply:
@spicydonuts Ah, then I think the example needs to be updated to use the constant matching format.